apps/microsoft-teams/frontend/src/hooks/useGetContentTypes.ts

```diff
@@ -18,17 +19,41 @@ const useGetContentTypes = () => {
   const getAllContentTypes = useCallback(async () => {
     try {
       setLoading(true);
-      // TODO: Implement pagination for content types
-      const contentTypesResponse = await sdk.cma.contentType.getMany({});
-      setAllContentTypes(contentTypesResponse.items || []);
+
+      const contentTypesSuperset: ContentTypeProps[] = [];
+      let skip = 0;
+      let needToFetchNextBatch = true;
+
+      while (needToFetchNextBatch) {
```

Comment:
I think a while loop is quite appropriate here.

For me (subjective opinion) it cleans things up / gets rid of a tiny bit of noise to embrace explicit control flow and ditch managing the extra mutable variable. Something like:

```js
while (true) {
  // ... fetch all the things and push
  if (contentTypesSuperset.length >= totalNumOfContentTypes) break;
  skip += CONTENT_TYPE_LIMIT;
}
```

Just an optional suggestion for your consideration.

Reply:
I could definitely go either way, I slightly prefer keeping the `needToFetchNextBatch`, just because it explains what the entry condition of the while loop is in english.  But, not a deal breaker.

I'm gonna keep it for now, but I appreciate the suggestion 👍